src/features/userMenu/UserMenu.tsx

```diff
@@ -0,0 +1,120 @@
+import styles from './user-menu.module.css'
+import { Button } from "~shared/ui/button";
+import { PAGE_PATH } from "~shared/config";
+import { Link } from "react-router-dom";
+import { sessionApi } from "~entities/session";
+import { useQueryClient } from "@tanstack/react-query";
+import { logout } from "~features/session";
+import { IUser } from "~shared/api";
+import { sessionKeys } from "~entities/session/api/sessionApi.ts";
+import { useRef, useState } from "react";
+import Icon from "~shared/ui/icon";
+import cn from "classnames";
+import { useClickOutside } from "~shared/lib/useClickOutside";
+import Navigation from "~entities/navigation";
+
+export function UserMenu() {
+  const queryClient = useQueryClient();
+  sessionApi.useCurrentUser();
+
+  const [ menuOpen, setMenuOpen ] = useState(false);
+  const [ user, setUser ] = useState<IUser | undefined>(undefined);
+
+
+  if (!user && queryClient.getQueryData(sessionKeys.session.currentUser())) {
+    setUser(queryClient.getQueryData(sessionKeys.session.currentUser()))
+  }
+
+  const logoutClick = async () => {
+    await logout(queryClient);
+    setUser(undefined)
```

Comment:
logout should remove the user from the store and the function above should handle this